crates/metaslang/bindings/src/lib.rs

```diff
@@ -22,32 +22,20 @@ type FileHandle = stack_graphs::arena::Handle<stack_graphs::graph::File>;
 type CursorID = usize;
 
 pub use location::{BindingLocation, BuiltInLocation, UserFileLocation};
-
-pub struct DefinitionHandle(GraphHandle);
-
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub(crate) enum Tag {
-    Alias,
-    C3,
-    Super,
-}
+pub use resolved::{BindingGraph, Definition, Reference};
 
 pub(crate) struct DefinitionBindingInfo<KT: KindTypes + 'static> {
-    definiens: Option<Cursor<KT>>,
-    tag: Option<Tag>,
+    definiens: Cursor<KT>,
     parents: Vec<GraphHandle>,
-    export_node: Option<GraphHandle>,
-    import_nodes: Vec<GraphHandle>,
     extension_scope: Option<GraphHandle>,
     inherit_extensions: bool,
 }
 
 pub(crate) struct ReferenceBindingInfo {
-    tag: Option<Tag>,
     parents: Vec<GraphHandle>,
 }
 
-pub struct BindingGraph<KT: KindTypes + 'static> {
+pub struct BindingGraphBuilder<KT: KindTypes + 'static> {
```

Comment:
I'm a bit confused by the hierarchy here:

- `BindingGraphBuilder` exposed from `lib.rs`, and is different than `Builder`, which is exposed from `builder/mod.rs`.
-  `BindingGraph` is exposed from `resolved/mod.rs`, and is different than `Graph`, which is exposed from `metaslang_graph_builder::graph`.

WDYT of restructuring it a bit to clarify the relationships between them? If I can suggest, ordering it by the public API/use cases:

- `builder/mod.rs` exposes the public `BindingGraphBuilder`:
  - Has the internal `Builder` and `Resolver` under it.
- `graph/mod.rs` exposes the public `BindingGraph`, and the related public APIs, like:
  - `graph/definition.rs`
  - `graph/reference.rs`
  - `graph/location.rs`

Not blocking for this PR of course.

Reply:
Yeah, I'm not happy about the module structure either. What we currently have in the `builder` module should probably be called `loader`, since it builds a graph and loads it into our stack graph. Then the `resolver` and `BindingGraphBuilder` could live in a `builder` module.

Reply:
I reorganized to code and put all the builder, resolver and loader code under a `builder` module.